### ISSUE TITLE
[recipe request] OpenAI Gym for Reinforcement Learning

### ISSUE THREAD

**Issue by mgiessing** [IBM/powerai#164]
Please add OpenAI Gym: https://gym.openai.com/

Name of package: OpenAI Gym
URL of package source code: https://github.com/openai/gym
License of package: The MIT License
List of package dependencies: Not known
Short description of package and use case(s): Needed to develop reinforcement learning models on IBM Power Hardware (AC922)

**Comment by hartb** [comment]
I think this maybe just went in yesterday:

```
$ conda search -c powerai gym                       
Loading channels: done                       
# Name                       Version           Build  Channel                             
gym                           0.15.3            py_0  powerai
```

FYI @npanpaliya

**Comment by jayfurmanek** [comment]
Yup. Already available! Thanks!

**Comment by mgiessing** [comment]
Sorry haven't looked into it recently, thanks a lot!
Just for communication to client - this is only for Python 2.7 correct?
At least it just installs for 2.7 on my kvm :)

**Comment by npanpaliya** [comment]
gym is a noarch package. It should work with py36 or py37 too.

**Comment by mgiessing** [comment]
Not sure if I'm doing anything wrong, but on 3.7 I get this error:

```
(wmlce-1.6.2) [root@p1201-kvm1 ~]# conda install -c powerai gym

UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

  - gym -> python[version='>=2.7,<2.8.0a0']

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to.  Your current python version
is (python=3.7).  Note that conda will not change your python version to a different minor version
unless you explicitly specify that.
```

If I understand that correctly this means gym is only available for Python 2.7-2.8?

**Comment by npanpaliya** [comment]
How did you create the environment?
I'm able to install it in py36 environment.

**Comment by mgiessing** [comment]
Mhh this is weird, I will write down the whole process:

[root@p1201-kvm1 ~]# wget Anaconda3-2019.07-Linux-ppc64le.sh
[root@p1201-kvm1 ~]# bash Anaconda3-2019.07-Linux-ppc64le.sh
[root@p1201-kvm1 ~]# source .bashrc
(base) [root@p1201-kvm1 ~]# conda config --prepend channels https://public.dhe.ibm.com/ibmdl/export/pub/software/server/ibm-ai/conda/
(base) [root@p1201-kvm1 ~]# conda create --name wmlce-1.6.2 python=3.7
(base) [root@p1201-kvm1 ~]# conda activate wmlce-1.6.2
(wmlce-1.6.2) [root@p1201-kvm1 ~]# conda install -c powerai gym

Still get the error :/

**Comment by mgiessing** [comment]
You install it via `conda install gym` , but this doesn't work for me.

```
(wmlce-1.6.2) [root@p1201-kvm1 ~]# conda install gym
Collecting package metadata (current_repodata.json): done
Solving environment: failed with current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - gym

Current channels:

  - https://public.dhe.ibm.com/ibmdl/export/pub/software/server/ibm-ai/conda/linux-ppc64le
  - https://public.dhe.ibm.com/ibmdl/export/pub/software/server/ibm-ai/conda/noarch
  - https://repo.anaconda.com/pkgs/main/linux-ppc64le
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/linux-ppc64le
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

**Comment by npanpaliya** [comment]
I installed using `conda install gym` because I'd added `powerai` in my channel list. Also, I found that gym installation works when you have conda-forge added in the channel list. In absense of conda-forge (with just defaults and powerai public channel), gym install isn't working and I too see the same  error as you're seeing. I'll investigate this.
@jayfurmanek - It might be satisfying some of its dependencies from conda-forge for sure. But I'm not sure if this is acceptable behavior or a problem to look at.
